test/tuple_algorithms.cpp

```diff
@@ -90,6 +90,25 @@ TEST_CASE("apply", "[tuple_algorithms]") {
     CHECK(t == stdx::tuple{2, 3, 4});
 }
 
+TEST_CASE("variadic apply", "[tuple_algorithms]") {
+    static_assert(stdx::apply([](auto... xs) { return (0 + ... + xs); }) == 0);
+    static_assert(stdx::apply([](auto... xs) { return (0 + ... + xs); },
+                              stdx::tuple{1, 2, 3},
+                              stdx::tuple{4, 5, 6}) == 21);
+
+    auto stateful = [calls = 0](auto...) mutable { return ++calls; };
+    CHECK(stdx::apply(stateful) == 1);
+    CHECK(stdx::apply(stateful) == 2);
+
+    static_assert(stdx::apply([](auto x, auto y) { return x.value + y.value; },
+                              stdx::tuple{move_only{42}},
+                              stdx::tuple{move_only{17}}) == 59);
+
+    auto t = stdx::tuple{1, 2, 3};
+    stdx::apply([](auto &...xs) { (++xs, ...); }, t, t);
+    CHECK(t == stdx::tuple{3, 4, 5});
```

Comment:
question: this is a really confusing test cause...took me several minutes to figure out what's going on. is there any way to make it more clear what it's intending to do? is it just checking that references to elements are getting passed correctly?

Reply:
Good call - improved it.